### ISSUE TITLE
False positive: possibly undefined loop variable

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#2039]
### Steps to reproduce
1. Run `pylint` over this minimal test case:

````
from random import randint
def example():
    for p in [8000, 8001, 8002, 8003]:
        if randint(0, 10) == 0:
            break
    print(p)
````

### Current behavior

```
W:  7,10: Using possibly undefined loop variable 'p' (undefined-loop-variable)
```
### Expected behavior

No warning should be emitted. `p` is guaranteed to be initialised to at least the first item in the list.

### pylint --version output

```
bje@bapbop:~$ pylint --version
Using config file /home/bje/.pylintrc
pylint 1.8.4, 
astroid 1.6.3
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609]
```

**Comment by bje-** [comment]
Thanks for the fast fix!
